Support for Conan and CMake

matthew-rindel opened the pull request.

matthew-rindel committed:
Added first set of changes to enable cross platform compiling (starting with Windows)

matthew-rindel committed:
README now has remote step

matthew-rindel committed:
Made everything compile and link using cmake and conan.

matthew-rindel committed:
Ignored more files

Added build folder to conanfile.py

matthew-rindel committed:
Updated README.md

matthew-rindel committed:
Removed some whitespace

Added bin to gitignore

GeekTeaa commented on README.md:
```diff
@@ -45,6 +45,21 @@ The goals of LangOCR are to do the following.
   operates.
 
 ## Installation
+First, add the bincrafters remote for Conan:
```
Please add a hyperlink to Conan documentation so people who are not enlightened can get and idea what Conan is.

matthew-rindel committed:
Updated docs to reflect conan documenation

Added more specific actions for build issues

Added workaround for build folder to get set correctly

matthew-rindel committed:
Added cmake as a build requirement

matthew-rindel committed:
Added code to clear out unneeded cmake files

GeekTeaa commented:
Declining PR cause I already merged it on accident via my git shenanigans. 

And it's been great. Thanks for your help! I see the appeal now for cmake and conan
